numba/core/types/abstract.py

```diff
@@ -1,4 +1,5 @@
 from abc import ABCMeta, abstractmethod, abstractproperty
+from typing import Dict as mpDict, Type as mpType
```

Comment:
`typing` isn't `mypy`, nor is `mypy` the only package that understands `typing`. `tDict` and `tType` would probably be better names.

Reply:
Thanks for looking at the PR @eric-wieser 
you're right, it's a choice I wasn't completely happy with and I was expecting that someone could mention it. Normally I would have used the word "typing", since as you say `mypy` is not `typing` nor the only static checker. The problem is that `typing` already has a meaning within Numba.
Your suggestion of using `t` as a prefix instead of `mp` makes sense, but I wonder if it could be confused with `typed.Dict`. 
I'll think about it a bit more, wait to see if anyone has another idea, and if nothing better comes up, I'll switch to `tDict` and `tType`.

Reply:
Could use `b` for builtin or `p` for python also.

Or perhaps `import typing as pt`, then use `pt.List` etc.

Reply:
I think `pt` would work well, thanks for the suggestion. I used it as a prefix though, what do you think? ie `ptDict` and `ptType` (latest commits has the updated names).
I'm also adding a note in the developer manual (`contributing` section), explaining this convention.